snuba/subscriptions/data.py

```diff
@@ -9,42 +9,59 @@
 from snuba.request.request_settings import SubscriptionRequestSettings
 from snuba.request.schema import RequestSchema
 from snuba.utils.metrics.timer import Timer
-from snuba.web.views import validate_request_content
 
 SUBSCRIPTION_REFERRER = "subscription"
 
 
+class InvalidSubscriptionError(Exception):
+    pass
+
+
 @dataclass(frozen=True)
 class Subscription:
     """
     Represents the state of a subscription.
     """
 
-    id: str
     project_id: int
     conditions: Sequence[Condition]
     aggregations: Sequence[Aggregation]
     time_window: timedelta
     resolution: timedelta
 
+    def __post_init__(self):
```

Comment:
```suggestion
    def __post_init__(self) -> None:
```

Reply:
Mypy never catches this, do I have something configured wrong?

Reply:
Hmm, do you not get this when running on c3b7c208cd6bc36b662b851b3edd0e3cfadf2767?

```
ted@veneno % python -m mypy --strict snuba/subscriptions | grep snuba/subscriptions
snuba/subscriptions/data.py:32: error: Function is missing a return type annotation
snuba/subscriptions/data.py:32: note: Use "-> None" if function does not return a value
```

Reply:
Yeah, I didn't have `--strict`, should be able to catch these going forward, thanks!